src/Data/V/Linear/Internal/Ambiguous.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# OPTIONS_HADDOCK hide #-}
+
+module Data.V.Linear.Internal.Ambiguous (theLength, make, FunN) where
+
+import Data.Kind
+import Data.Proxy
+import Data.Type.Equality
+import {-# SOURCE #-} Data.V.Linear.Internal.V (V (..), cons)
```

Comment:
Let's not. I really don't want recursive imports unless it's absolutely necessary. I'd rather you add extra files instead.

Though, I guess this file is temporary? Until we clean up `make`? If so, I guess it's ok.

Reply:
Yeah, the goal is just to have most of the `V` functions in a file without the `AllowAmbiguousTypes` extension.

Reply:
Mmm… but `make` will always have ambiguous type variables, most likely. So we will have this problem. I'd really like to avoid recursive imports.